### PR TITLE
IndexScanDesc and IndexScanPlan Refactoring

### PR DESCRIPTION
**Pull request by malin1993ml**
This PR fixes the issue in #1300. It takes the pointer out of the `IndexScanDesc` and `IndexScanPlan`, and replace them with the index OID. The `IndexScanExecutor` will get the physical pointer to the index object in its `DInit()` function.

This PR also adjusts the calls to the `IndexScanDesc` constructor in many test cases.

### CODE REVIEW

File: src/planner/index_scan_plan.cpp [hunk 1]
```diff
@@ -55,8 +55,8 @@ IndexScanPlan::IndexScanPlan(storage::DataTable *table,
   //
   // Values that are left blank will be recorded for future binding
   // and their offset inside the value array will be remembered
-  index_predicate_.AddConjunctionScanPredicate(index_.get(), values_,
-                                               key_column_ids_, expr_types_);
+  //index_predicate_.AddConjunctionScanPredicate(index_.get(), values_,
+  //                                             key_column_ids_, expr_types_);
```
Comment (sivaprasadsudhir) [inline src/planner/index_scan_plan.cpp]: Why don't we delete these lines?

Comment (malin1993ml) [reply src/planner/index_scan_plan.cpp]: Goot catch! I forgot to do delete them. I'll do it.

Comment (vkonagar) [reply src/planner/index_scan_plan.cpp]: If we are shifting this from the planner to the execution engine, we won't be able to see the details of these predicates at the optimizer side, right? The plan returned by the optimizer to 'What-if index' will not have the required index plan details?

**Comment by apavlo** [comment]
I will merge this when the build passes.

**Pull request merged**